javascript/javascript2/week2/workingWithMovies.js

```diff
@@ -0,0 +1,93 @@
+// 1. Array containing the movies with a short title
+
+const shortTitleMovies = movies.filter(movie => movie.title.length < 4);
+console.log(shortTitleMovies);
+
+
+// 2. Array of movie titles with long movie titles
+
+const longTitleMovies = movies.filter(movie => movie.title.length > 45);
+console.log(longTitleMovies);
+
+
+// 3. Count the number of movies made between 1980-1989
+
+const countByYear = movies.filter(movie => movie.year >= 1980 && movie.year <= 1989).length;
+console.log(countByYear);
+
+
+/* 4. Create a new array that has an extra key called tag.
+*The tag is based on the rating: Good (>= 7), Average (>= 4 and < 7), Bad (< 4)*/
+
+const ratingTag = movies.map(movie => {
+    movie["tag"] = movie.rating >= 7 ? "Good" :
+        movie.rating >= 4 && movie.rating < 7 ? "Average" : "Bad";
+    return movie;
+})
+console.log(ratingTag);
+
+
+/* 5.Using chaining, first filter the movies array to only contain 
+*the movies rated higher than 6. 
+*Now map the movies array to only the rating of the movies.*/
+
+const moviesRating = movies.filter(movie => movie.rating > 6).map(movie => `Rating : ${movie.rating}`);
+console.log(moviesRating);
+
+// 6.Count the total number of movies containing the specific words in their title.
+
+
+// First solution
+
+const moviesTitle = movies.map(movie => {
+    let surferTitle = (movie.title.match(/Surfer/gi) || []).length;
+    let alienTitle = (movie.title.match(/Alien/gi) || []).length;
+    let benjaminTitle = (movie.title.match(/Benjamin/gi) || []).length;
+    let Count = surferTitle + alienTitle + benjaminTitle;
+    return Count;
+});
+const TotalCount = moviesTitle.reduce((sum, num) => sum + num);
+console.log(TotalCount);
+
+// Second solution
+let moviesCount = movies.filter(movie => movie.title.toLowerCase().includes("Surfer".toLowerCase()) ||
```

Comment:
```suggestion
let moviesCount = movies.filter(movie => movie.title.toLowerCase().includes("surfer") ||
```

Reply:
Hi Cameron, 
Thank you for your time and help.
I have a question. when I try to delete the .toLowerCase ,as you said here. The result will be 0 instead of 19. It seems that we need this .toLowerCase both after title and after the keyword to get the result of 19. But I do not know why we need both of the.